pkg/cli/clusteroperator/copytonode/command.go

```diff
@@ -0,0 +1,90 @@
+package copytonode
+
+import (
+	"github.com/openshift/oc/pkg/cli/clusteroperator/pernodepod"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	copyToNodeLong = templates.LongDesc(`
+		Copies file from the host to the specified nodes.
+
+		Experimental: This command is under active development and may change without notice.
+	`)
+
+	copyToNodeExample = templates.Examples(`
+		# copy a new bootstrap kubeconfig file to every node
+		oc adm clusteroperators copy-to-node --copy=new-bootstrap-kubeconfig=/etc/kubernetes/kubeconfig
+
+		# copy a new bootstrap kubeconfig file to masters
+		oc adm clusteroperators copy-to-node --copy=new-bootstrap-kubeconfig=/etc/kubernetes/kubeconfig -l node-role.kubernetes.io/master`)
+)
+
+type CopyToNodeOptions struct {
+	PerNodePodOptions *pernodepod.PerNodePodOptions
+
+	// FileSources to derive the secret from (optional)
+	FileSources []string
+
+	genericclioptions.IOStreams
+}
+
+func NewRestartKubelet(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *CopyToNodeOptions {
+	return &CopyToNodeOptions{
+		PerNodePodOptions: pernodepod.NewPerNodePodOptions(
+			"openshift-copy-to-node-",
+			restClientGetter,
+			streams,
+		),
+
+		IOStreams: streams,
+	}
+}
+
+func NewCmdCopyToNode(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewRestartKubelet(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   "copy-to-node",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Copies specified files to the node."),
```

Comment:
There's actually `oc rsync` today, it just doesn't support `oc rsync foo node:`.

(And a bunch of backing infrastructure there that maybe we could share some with)

Reply:
> There's actually oc rsync today, it just doesn't support oc rsync foo node:.
>
> (And a bunch of backing infrastructure there that maybe we could share some with)

Doing a single pod versus doing large numbers of nodes presents a different problem.  Rather than setting up oc client to pod connections for every single node, this creates an intermediate secret which is then mounted, so the network connection remains internal to the cluster.